jax/interpreters/pxla.py

```diff
@@ -298,29 +280,22 @@ class ShardedDeviceArray(xla.DeviceArray):
   that is, it is only an optimization to reduce transfers.
 
   The number of device buffers underlying a ShardedDeviceArray instance is equal
-  to the number of replicas of the computation that produced it. Each buffer
-  represents a shard of the original array, meaning a slice along its leading
-  axis. These component buffers reside on distinct devices, but need not
-  represent distinct logical shards. The correspondence can be computed with
-  the assign_shards_to_replicas function.
+  to the number of devices of the computation that produced it. Each buffer
+  represents a shard of the original array, indicated by its corresponding index
```

Comment:
Can you better document what `logical_indices` means? I couldn't quite follow from reading the comments.

Reply:
I added a description of the logical_indices field. Please let me know if it's still unclear or you have any suggestions for improvement.

Reply:
Before you updated it, I was going to comment with another one that might work here and/or in the `shard_args` docstring (I'm using the `indexers` name):
"Each buffer has a corresponding indexer in the `indexers` list, and represents the same portion of the overall array as the indexer would in standard NumPy indexing. That is, a buffer with indexer `(3, slice(None))` represents the same part of an array as `arr[3, :]` does. It is possible for multiple buffers to represent the same portion of a `ShardedDeviceArray`; in that case, the buffers contain the same values but reside on distinct devices."
Your new description is better, though, I think!

Reply:
I changed it around some more based on your suggestion. Hopefully this + the below attributes section is clear now.